src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+
+        hetPriorType = HetPriorType.HETEROGENEOUS;
+
+        /* parameters for building the allele ratio prior at Het sites */
+        this.minAbnormalFraction = ParamUtils.inRange(minAbnormalFraction, 0.0, 1.0, "Minimum fraction of abnormal" +
+                " cells must be between 0 and 1.");
+        this.maxAbnormalFraction = ParamUtils.inRange(maxAbnormalFraction, this.minAbnormalFraction, 1.0, "Maximum fraction of abnormal" +
+                " cells must be greater than the provided minimum and less than 1.");
+        this.maxCopyNumber = ParamUtils.isPositive(maxCopyNumber, "Maximum copy number must be positive");
+
+        /* auxiliary member functions */
+        this.minHetAlleleFraction = (1 - this.maxAbnormalFraction) / (this.maxCopyNumber * this.maxAbnormalFraction +
+                2 * (1 - this.maxAbnormalFraction));
+        this.breakpointHetAlleleFraction = (1 - this.minAbnormalFraction) / (this.maxCopyNumber * this.minAbnormalFraction +
+                2 * (1 - this.minAbnormalFraction));
+
+        /* initialize the integration quadrature and calculate the allele fraction prior on the abscissas */
+        initializeIntegrationQuadrature(ParamUtils.isPositive(quadratureOrder - MIN_QUADRATURE_ORDER,
+                "Quadrature order must be greater than " + MIN_QUADRATURE_ORDER) + MIN_QUADRATURE_ORDER);
+        initializeHetAlleleFractionPrior();
+        logger.info("Using the HETEROGENEOUS Het prior.");
+    }
+
+    /******************************
+     * core computational methods *
+     ******************************/
+
+    /**
+     * Initilizes the quadrature for calculating allele ratio integrals in getHetLogLikelihood
+     * @param numIntegPoints  number of points in the quadrature
+     */
+    private void initializeIntegrationQuadrature(final int numIntegPoints) {
+
+        /* get Gauss-Legendre quadrature factory of order @numIntegPoints */
+        GaussIntegratorFactory integratorFactory = new GaussIntegratorFactory();
+        GaussIntegrator gaussIntegrator = integratorFactory.legendre(numIntegPoints,
+                minHetAlleleFraction, 1.0 - minHetAlleleFraction);
+
+        /* abscissas */
+        gaussIntegrationAbscissas.clear();
+        gaussIntegrationAbscissas.addAll(IntStream.range(0, numIntegPoints).
+                mapToDouble(gaussIntegrator::getPoint).boxed().collect(Collectors.toList()));
+
+        /* weights */
+        gaussIntegrationWeights.clear();
+        gaussIntegrationWeights.addAll(IntStream.range(0, numIntegPoints).
+                mapToDouble(gaussIntegrator::getWeight).boxed().collect(Collectors.toList()));
+
+        /* log of weights */
+        gaussIntegrationLogWeights.clear();
+        gaussIntegrationLogWeights.addAll(gaussIntegrationWeights.stream().
+                mapToDouble(FastMath::log).boxed().collect(Collectors.toList()));
+    }
+
+    /**
+     * (advanced) calculate a simple prior (and log prior) for the allele fraction based on
```

Comment:
What does ```(advanced)``` means? that I shouldn't feel bad if I don't understand what it means?.